### PR TITLE
document macros

### PR DESCRIPTION
**Pull request by oberblastmeister**

### CODE REVIEW

File: helix-view/src/macros.rs [hunk 1]
```diff
@@ -1,3 +1,14 @@
+//! These are macros to make getting very nested fields in the `Editor` struct easier
```
Comment (kirawi) [inline helix-view/src/macros.rs:1]: You should end the sentences with a period.

File: helix-view/src/macros.rs [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+//! These are macros instead of functions because functions will have to take `&mut self`
+//! However, rust doesn't know that you only want a partial borrow instead of borrowing the
+//! entire struct which `&mut self` says.  This makes it impossible to do other mutable
```
Comment (kirawi) [inline helix-view/src/macros.rs:4]: Double space after a period.

File: helix-view/src/macros.rs [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+//! stuff to the struct because it is already borrowed. Because macros are expanded,
+//! this circumvents the problem because it is just like indexing fields by hand and then
+//! putting a &mut in front of it. This way rust can see that we are only borrowing a
```
Comment (kirawi) [inline helix-view/src/macros.rs:7]: rust -> Rust

**Review by kirawi — CHANGES_REQUESTED** [review]
I would simplify the block of text into something like:
`//! Macros to partially borrow \`Editor\`, until https://github.com/rust-lang/rfcs/issues/1215.`

I think that as it is currently, it's very confusing and is over explaining what these macros used for.

File: helix-view/src/macros.rs (hunk 3 above)
Comment (pickfire) [inline helix-view/src/macros.rs:7]:
```suggestion
//! putting a `&mut` in front of it. This way rust can see that we are only borrowing a
```

**Review by archseer — APPROVED** [review]

**Pull request merged**